### PR TITLE
Record Sidekiq stats with collectd

### PR DESCRIPTION
**Pull request by alecgibson**
In order to help diagnose excessive memory usage of worker tasks,
this change collects Sidekiq process data with collectd, for monitoring
in Graphite.

worker-rss-usage

### CODE REVIEW

**Comment by alecgibson** [comment]
[Memory issues in Sidekiq jobs](https://trello.com/c/I5Nz2FeO/408-memory-issues-in-sidekiq-jobs)

File: modules/govuk/manifests/procfile/worker.pp [hunk 1]
```diff
@@ -66,6 +67,10 @@
         service_description => "${title} procfile worker upstart up",
         host_name           => $::fqdn,
       }
+      collectd::plugin::process { "app-worker-${title_underscore}":
+        ensure => $ensure,
+        regex  => "sidekiq .* ${title}.*\\.gov\\.uk",
+      }
```
Comment (cbaines) [inline modules/govuk/manifests/procfile/worker.pp]: If you put this block outside the `if $ensure == present {`, the use of `$ensure` in the block will mean that the `collectd::plugin::process` thing is removed if `$ensure` specifies it should be. At the moment, `$ensure` will always be `present` here, as that is the condition in the `if`.

**Comment by alecgibson** [comment]
@cbaines does this look alright now?

**Comment by cbaines** [comment]
> @cbaines does this look alright now?

Yep, I think that will work better.

**Pull request merged**